src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -327,15 +327,18 @@ instance Buildable TxStatus where
 
 instance NFData TxStatus
 
+-- | The flow of funds in to or out of a wallet.
 data Direction
-    = Outgoing
-    | Incoming
+    = Outgoing -- ^ Funds exit the wallet.
+    | Incoming -- ^ Funds enter the wallet.
+    | Internal -- ^ Payment from the wallet to itself.
```

Comment:
Yeah, I was wondering yesterday if we couldn't give this one a special semantic :/ I was going for a simpler / less granular semantic where `Outgoing` just means not `Incoming` and `Incoming` means that there's no inputs from the wallet itself. So a transaction from the wallet to the wallet would effectively be labelled as `Outgoing`. I think it's nice to be slightly more granular here with the `Internal`.